src/fprime/fbuild/interaction.py

```diff
@@ -216,3 +296,200 @@ def new_component(
     except OSError as ose:
         print("[ERROR] {}".format(ose))
     return 1
+
+
+def is_valid_name(word: str):
+    invalid_characters = [
+        "#",
+        "%",
+        "&",
+        "{",
+        "}",
+        "/",
+        "\\",
+        "<",
+        ">",
+        "*",
+        "?",
+        " ",
+        "$",
+        "!",
+        "'",
+        '"',
+        ":",
+        "@",
+        "+",
+        "`",
+        "|",
+        "=",
+    ]
+    for char in invalid_characters:
+        if isinstance(word, str) and char in word:
+            return char
+        elif not isinstance(word, str):
+            raise ValueError("Incorrect usage of is_valid_name")
+    return "valid"
+
+
+def get_valid_input(prompt):
+    valid_name = False
+    while not valid_name:
+        name = input(prompt)
+        char = is_valid_name(name)
+        if char != "valid":
+            print("'" + char + "' is not a valid character")
+        else:
+            valid_name = True
+    return name
+
+
+def get_port_input(namespace):
+    # Gather inputs to use as context for the port template
+    defaults = {
+        "port_name": "ExamplePort",
+        "short_description": "Example usage of port",
+        "dir_name": ".",
+        "namespace": namespace,
+        "arg_list": [],
+    }
+    args_done = False
+    arg_list = []
+    port_name = get_valid_input("Port Name [{}]: ".format(defaults["port_name"]))
+    short_description = input(
+        "Short Description [{}]: ".format(defaults["short_description"])
+    )
+    dir_name = get_valid_input("Directory Name [{}]: ".format(defaults["dir_name"]))
+    namespace = get_valid_input("Port Namespace [{}]: ".format(defaults["namespace"]))
+    while not args_done:
+        if arg_list == []:
+            add_arg = confirm("Would you like to add an argument?: ")
+        else:
+            add_arg = confirm("Would you like to add another argument?: ")
+        if add_arg:
+            arg_name = get_valid_input("Argument name: ")
+            arg_type = get_valid_input(
+                "Argument type (Valid primitive types are: BOOL, I8, I16, "
```

Comment:
BOOL actually doesn't look like it's a valid F' type. Not sure how it wound up in the docs. It can probably removed from this list, at least for now